### ISSUE TITLE
Profit margin

### ISSUE THREAD

**Issue by calamarcus** [abayomi185/simple-pump-and-dump-bot#36]
Hi again. How is the profit function work. i went live on a pump and put 0.5 profit in the config. i finished woth a profit, but at 8%, so im eager to learn how the bot works in regards to getting the most profit.

**Comment by nikeboy264** [comment]
There are 2 reasons:
1. The bot buy/sell using market order, not limit. (The market price always less than or equal avg price)
2. Because sell using market order, so the bot keep checking current avg price, if it reaches the profit then the bot sell the coin (it take a little time from 'check' to 'sell'). In a pump game, the avg price up/down very quick, so at the time that the avg price reaches the profit, then maybe at the moment the bot does sell, the avg price goes down.

**Comment by calamarcus** [comment]
Okay. That is not good. An improved solution might be needed.

**Comment by abayomi185** [comment]
Hi @nikeboy264 you seem knowledgeable on this. Would it be beneficial to to be able to make limit sell order too? Or would that be too risky when selling?

**Comment by rishabhkr432** [comment]
I have a basic algorithm, I hope this helps.
2. Set a limit-sell at the start. 
3. Store the clientOid or orderID
4. Check if limit-sell is filled or not using isActive in Data dict.


`


"data": {      "id": "5c35c02703aa673ceec2a168",      "symbol": "BTC-USDT",      "opType": "DEAL",      "type": "limit",      "side": "buy",      "price": "10",      "size": "2",      "funds": "0",      "dealFunds": "0.166",      "dealSize": "2",      "fee": "0",      "feeCurrency": "USDT",      "stp": "",      "stop": "",      "stopTriggered": false,      "stopPrice": "0",      "timeInForce": "GTC",      "postOnly": false,      "hidden": false,      "iceberg": false,      "visibleSize": "0",      "cancelAfter": 0,      "channel": "IOS",      "clientOid": "",      "remark": "",      "tags": "",      **"isActive": false**,      "cancelExist": false,      "createdAt": 1547026471000,      "tradeType": "TRADE"    }

`

6.  If limit-sell is not filled within the set time then cancel the limit order using the stored orderID and create  new a market sell order.

**Comment by nikeboy264** [comment]
> Hi @nikeboy264 you seem knowledgeable on this. Would it be beneficial to to be able to make limit sell order too? Or would that be too risky when selling?

I agree with @rishabhkr432's solution.
To lower the risk and more advanced, i think:
1. Buy the coin with Market order. Timer to sell fallback now activated.
2. Split the coin qty to 4 parts. (E.g bot was bought 100 BTC)
3. Set the profit expected for each part (in configuration file). E.g:
1st 25BTC - Profit margin: 10%
2nd 25BTC - Profit margin: 20%
3rd 25BTC - Profit margin: 50%
4th 25BTC - Profit margin: 80%

4. Place 4 sell orders using Limit order. Limit order is the best because when the coin price reach profit expected, the coin will be sold immediately.
5. I think the bot no need to waste time to check if the Limit sell order is filled. If Timer (in step 1) is end, the bot cancelled all available Limit Sell order and do sell fallback.

**Comment by rishabhkr432** [comment]
> > Hi @nikeboy264 you seem knowledgeable on this. Would it be beneficial to to be able to make limit sell order too? Or would that be too risky when selling?
> 
> I agree with @rishabhkr432's solution.
> To lower the risk and more advanced, i think:
> 
> 1. Buy the coin with Market order. Timer to sell fallback now activated.
> 2. Split the coin qty to 4 parts. (E.g bot was bought 100 BTC)
> 3. Set the profit expected for each part (in configuration file). E.g:
>    1st 25BTC - Profit margin: 10%
>    2nd 25BTC - Profit margin: 20%
>    3rd 25BTC - Profit margin: 50%
>    4th 25BTC - Profit margin: 80%
> 4. Place 4 sell orders using Limit order. Limit order is the best because when the coin price reach profit expected, the coin will be sold immediately.
> 5. I think the bot no need to waste time to check if the Limit sell order is filled. If Timer (in step 1) is end, the bot cancelled all available Limit Sell order and do sell fallback.

Yeah, this sounds more reasonable. Checking limit-sell again would be time-consuming.

**Comment by sNnickK** [comment]
What is the best settings for huge pumps like 1000% ?

**Comment by rishabhkr432** [comment]
Lol bruh your expectations seem to be really high. Calm down. 
Also I replied to you in the issue.
